substrate/frame/stake-tracker/src/lib.rs

```diff
@@ -0,0 +1,347 @@
+};
+use sp_std::collections::btree_map::BTreeMap;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+/// The balance type of this pallet.
+pub type BalanceOf<T> = <<T as Config>::Staking as StakingInterface>::Balance;
+/// The account ID of this pallet.
+pub type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+
+/// Represents a stake imbalance to be applied to a staker's score.
+#[derive(Copy, Clone, Debug)]
+pub enum StakeImbalance<Balance> {
+	// Represents the reduction of stake by `Balance`.
+	Negative(Balance),
+	// Represents the increase of stake by `Balance`.
+	Positive(Balance),
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use crate::*;
+	use frame_election_provider_support::VoteWeight;
+	use frame_support::pallet_prelude::*;
+
+	/// The current storage version.
+	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Currency: Currency<Self::AccountId, Balance = BalanceOf<Self>>;
+
+		/// The staking interface.
+		type Staking: StakingInterface<AccountId = Self::AccountId>;
+
+		/// Something that provides a best-effort sorted list of voters.
+		///
+		/// To keep the load off the chain as much as possible, changes made to the staked amount
+		/// via rewards and slashes are dropped and thus need to be manually updated through
+		/// extrinsics. In case of `bags-list`, this always means using `rebag` and `putInFrontOf`.
+		type VoterList: SortedListProvider<Self::AccountId, Score = VoteWeight>;
+
+		/// Something that provides a best-effort sorted list of targets.
+		///
+		/// Note that while at the time of nomination all targets are checked to be real
+		/// validators, they can chill at any point, and their approval stakes will still be
+		/// recorded. This implies that what comes out of iterating this list MIGHT NOT BE AN ACTIVE
+		/// VALIDATOR.
+		type TargetList: SortedListProvider<Self::AccountId, Score = VoteWeight>;
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Returns the vote weight of a staker based on its current *active* stake, as returned by
+		/// the staking interface.
+		pub(crate) fn active_vote_of(who: &T::AccountId) -> VoteWeight {
+			T::Staking::stake(who)
+				.map(|s| Self::to_vote(s.active))
+				.defensive_unwrap_or_default()
+		}
+
+		/// Converts a staker's balance to its vote weight.
+		pub(crate) fn to_vote(balance: BalanceOf<T>) -> VoteWeight {
+			<T::Staking as StakingInterface>::CurrencyToVote::to_vote(
+				balance,
+				T::Currency::total_issuance(),
+			)
+		}
+
+		/// Updates a staker's score by increasing/decreasing an imbalance of the current score in
+		/// the list.
+		pub(crate) fn update_score<L>(who: &T::AccountId, imbalance: StakeImbalance<VoteWeight>)
+		where
+			L: SortedListProvider<AccountIdOf<T>, Score = VoteWeight>,
```

Comment:
This is always called with `TargetList`, not sure it is generic?

Reply:
Yes, I just removed all comments and code related to the stake tracker.